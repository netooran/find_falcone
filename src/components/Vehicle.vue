<template>
  <div
    class="vehicle"
    :class="{active: isActive, selected: isSelected}"
    v-on:click="isActive && selectVehicle($vnode.key)"
  >
    <i class="fas fa-space-shuttle fa-4x"/>
    <div class="vehicle-info">
      <h4>{{ vehicle.name }} ({{ vehicle.total_no }})</h4>
      <h6>Speed: {{ vehicle.speed }} au/h</h6>
      <h6>Max distance: {{ vehicle.max_distance }} au</h6>
    </div>
  </div>
</template>

<script>
import { mapState, mapActions } from "vuex";

export default {
  name: "Vehicle",
  props: {
    vehicle: {
      name: String,
      total_no: Number,
      max_distance: Number,
      speed: Number
    }
  },
  methods: mapActions(["selectVehicle"]),
  computed: {
    ...mapState(["selectedPlanet", "squad"]),

    isActive() {
      return this.vehicle.total_no > 0;
    },

    isSelected() {
      return this.squad[this.selectedPlanet] == this.$vnode.key;
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.vehicle {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 30px;
  border-radius: 10px;
}

.active {
  box-shadow: 1px 1px 1px 1px #ccc;
}

.selected {
  color: #89c0b7;
}

.selected,
.active:hover {
  box-shadow: 1px 1px 3px 4px rgb(155, 154, 154);
}

.vehicle-info > * {
  margin: 0%;
}
</style>
