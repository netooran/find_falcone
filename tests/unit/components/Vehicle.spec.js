import { shallowMount } from "@vue/test-utils";
import Vehicle from "@/components/Vehicle.vue";

describe("Vehicle", () => {
  let wrapper, vehicle;

  beforeEach(() => {
    vehicle = { name: "BFR", total_no: 2, speed: 2, max_distance: 437 };
    wrapper = shallowMount(Vehicle, { propsData: { vehicle } });
  });

  it("renders name, total no, speed and max distance when passed", () => {
    expect(wrapper.text()).toMatch(vehicle.name);
    expect(wrapper.text()).toMatch(vehicle.total_no.toString());
    expect(wrapper.text()).toMatch(`${vehicle.speed} au/h`);
    expect(wrapper.text()).toMatch(`${vehicle.max_distance} au`);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
