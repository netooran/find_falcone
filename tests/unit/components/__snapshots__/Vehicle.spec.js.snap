// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicle has the expected html structure 1`] = `
<div
  class="vehicle"
>
  <i
    class="fas fa-space-shuttle fa-4x"
  />
   
  <div
    class="vehicle-info"
  >
    <h4>
      BFR (2)
    </h4>
     
    <h6>
      Speed: 2 au/h
    </h6>
     
    <h6>
      Max distance: 437 au
    </h6>
  </div>
</div>
`;
